extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -243,6 +246,26 @@ export class AmpAutocomplete extends AMP.BaseElement {
     });
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const src = mutations['src'];
+    if (src === undefined || src === null) {
+      return Promise.resolve;
+    }
+    if (typeof src === 'string') {
+      return this.getRemoteData_().then(value => {
+        this.inlineData_ = value;
```

Comment:
I think the meaning of the `inlineData_` property is getting confusing, since it is assigned the value from `getRemoteData_` which is not inline in the way that I understand inline. Also consider passing the data as a parameter to `renderResults_`, since it's an implicit dependency that might be better as explicit.

Reply:
This is a good point. I renamed inlineData_ to sourceData_ and tried to replace all references to variables named "data" with more clear naming, i.e. "remoteData", "filteredData".

I also added a method `filterDataAndRenderResults_` to take make the original intention of `renderResults_` the more readable, and refactored `renderResults_` to handle only the job of rendering filtered data in the dom. For both of these methods I tried to specify their implicit dependencies as explicit parameters. :) Let me know if this makes things more clear.